bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/processors/checksum/ChecksumUtilities.java

```diff
@@ -115,7 +115,8 @@ public static IStatus calculateChecksums(File pathOnDisk, Map<String, String> ch
 		for (IConfigurationElement checksumVerifierConfiguration : ChecksumUtilities
 				.getChecksumComparatorConfigurations()) {
 			String id = checksumVerifierConfiguration.getAttribute("id"); //$NON-NLS-1$
-			if (checksumsToSkip.contains(id))
+			if (checksumsToSkip.contains(id)
+					|| !Boolean.parseBoolean(checksumVerifierConfiguration.getAttribute("publish"))) //$NON-NLS-1$
```

Comment:
Does getAttribute return the default value when the attribute is missing from the plugin.xml?

Reply:
I don't think it does. `getAttribute` has no knowledge of the `.exsd` and the default value in exsd is supposed to reflect the default value implemented by consumers. So this condition would most likely lead to about no checksum being produced unless they explicitly set `publish=true`.

Reply:
As we have a test-case for this lets see if that already fails.

Reply:
> As we have a test-case for this lets see if that already fails.

It does: `ProduceChecksumTest.testChecksums:45 SHA256 was not computed!` lets adjust this ... seems there are some test for md5 that need to be removed/adjusted.